kraken-std/src/kraken/std/python/pyproject.py

```diff
@@ -51,8 +52,66 @@ def save(self, path: Path | None = None) -> None:
         with path.open("wb") as fp:
             tomli_w.dump(self.to_json(), fp)
 
-    def get_poetry_sources(self) -> list[dict[str, Any]]:
-        return list(self._poetry_section().setdefault("source", []))
+    def set_core_metadata_version(self, version: str | None) -> str | None:
+        """Updates the core metadata version field and returns the previous value"""
+        return self._set_version(self.setdefault("project", {}), version)
+
+    @staticmethod
+    def _set_version(config: Any, version: str | None) -> str | None:
+        """Updates the core metadata version field and returns the previous value"""
+        old_version = config.get("version")
+        if version is None:
+            if "version" in config:
+                del config["version"]
+        else:
+            config["version"] = version
+        return old_version  # type: ignore[no-any-return]
+
+    def _tool_section(self, name: str) -> Dict[str, Any]:
+        return self.setdefault("tool", {}).setdefault(name, {})  # type: ignore[no-any-return]
+
+    def _get_pyproj_sources(self, sources_conf: Dict[str, Any]) -> list[dict[str, Any]]:
+        return list(sources_conf.setdefault("source", []))
+
+    def _delete_pyproj_source(self, sources_conf: list[dict[str, Any]], source_name: str) -> None:
+        index = next((i for i, v in enumerate(sources_conf) if v["name"] == source_name), None)
+        if index is None:
+            raise KeyError(source_name)
```

Comment:
slightly more readable (IMO):

```suggestion
        index = [conf["name"] for conf in sources_conf].index(source_name)
```

(it will throw a `ValueError` instead of a `KeyError`, but I don't think it's that important here)

Reply:
fix